### DISCUSSION TITLE
Getting Changes from Unit of Work returns empty

### DISCUSSION THREAD

**Discussion by ordepahnuc** [mikro-orm/mikro-orm#5039] [Q&A] [answered]
Hello,

I'm using MikroORM v5.9 and after the version upgrading from v4.5, I'm unable to get the changes from the UoW after persist and before flush calls. Above follows an image with my current code.
![image](https://github.com/mikro-orm/mikro-orm/assets/55294477/ca22d375-8aed-4035-bb77-74684b3ec1cc)

I'm using this approach because I need to log every update and the person responsible for it. On this new version there is other approach that can solve my issue?

Best regards,
Pedro Cunha

**Comment by B4nan** [comment]
why are you calling `computeChangeSets`? its internal method that is called during flush, you are not supposed to call that anywhere in your code.

anyway, if you don't get any changes there, it simply means no changes were detected. but lets get back to what and why are you doing, as this feels wrong in the first place.

**Reply by ordepahnuc** [reply]
I want to grab the changes of entities (old value + new value), so I can registry the changes and who did them, like an Activity History.

By checking the documentation I can have the changes by creating subscribers, but then I lose the person who performed the update.

**Reply by B4nan** [reply, marked as answer]
you should be indeed using `onFlush` event for this, you will get the context (the EM fork) there, which you can feed with any data in some middleware (e.g. pass in the current user id) - you can leverage the filter parameters for this, as thats basically a generic storage for anything on the EM fork.

https://mikro-orm.io/docs/filters#global-filters

you can use the `em.setFilterParams()` in the middleware and `em.getFilterParams()` in the subscriber

**Reply by B4nan** [reply]
moreover, in v6, you can also leverage custom logger context which can be also set on the EM fork, so if this is just about logging, it might be cleaner approach

https://mikro-orm.io/docs/next/logging#providing-additional-context-to-a-custom-logger

the logger context can be set when forking or when creating context via `RequestContext` helper. i see we dont have a getter for the `em.loggerContext` which is now protected, will add one so it can be also used this way
